assets/js/components/data.js

```diff
@@ -610,17 +595,21 @@ const dataAPI = {
 	 * @return {Promise} A promise for the fetch request.
 	 */
 	set( type, identifier, datapoint, data ) {
-		if ( googlesitekit[ type ] && googlesitekit[ type ][ identifier ] ) {
-			googlesitekit[ type ][ identifier ][ datapoint ] = data;
-		}
-
 		const body = {};
 		body.data = data;
 
 		// Make an API request to store the data.
 		return wp.apiFetch( { path: `/google-site-kit/v1/${ type }/${ identifier }/data/${ datapoint }`,
 			data: body,
 			method: 'POST',
+		} ).then( ( response ) => {
+			dataAPI.invalidateCacheGroup( type, identifier, datapoint );
+
+			return new Promise( ( resolve ) => {
+				resolve( response );
+			} );
+		} ).catch( ( err ) => {
+			return Promise.reject( err );
```

Comment:
I think this `.catch` is redundant. Returning a rejected promise with the same error that was caught from the former should be the same behavior without it.